### DISCUSSION TITLE
Design Definitions

### DISCUSSION THREAD

**Discussion by arantespp** [TriangulosTecnologia/lps#3] [Ideas]
### Description

I want to test some theme definitions in our projects about font sizes and spacing. What is your opinion about them?

#### Font sizes

I want to follow [Tailwind's pattern](https://tailwindcss.com/docs/font-size) about font sizes. So my suggestion here is to use the following values in our theme files:

```js
  fontSizes: {
    xs: '0.75rem',
    sm: '0.875rem',
    base: '1rem',
    lg: '1.125rem',
    xl: '1.25rem',
    '2xl': '1.5rem',
    '3xl': '1.875rem',
    '4xl': '2.25rem',
    '5xl': '3rem',
    '6xl': '3.75rem',
    '7xl': '4.5rem',
    '8xl': '6rem',
    '9xl': '8rem',
  },
  lineHeights: {
    xs: '1rem',
    sm: '1.25rem',
    base: '1.5rem',
    lg: '1.75rem',
    xl: '1.75rem',
    '2xl': '2rem',
    '3xl': '2.25rem',
    '4xl': '2.5rem',
    '5xl': '1',
    '6xl': '1',
    '7xl': '1',
    '8xl': '1',
    '9xl': '1',
  },
```

With this configuration, we can add responsiveness fonts size in our projects with the key instead of array index: `fontSize: ['sm', 'lg']`.

#### Space

About [space array](https://theme-ui.com/theme-spec), I want to test out the values as Fibonacci series. These values give us a good range to use in our spacing properties (padding, margin, ...).

```js
  space: [
    0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597,
  ],
```

**Comment by enniolopes** [comment]
Nice, and the default root font size will be 16px (I mean if it's not defined at the project)?

Do you think we need to repeat the 1 of Fibonacci series?

**Reply by arantespp** [reply]
Generally, the default is 16px, but the user may want to change the default because it's easier for them, and our site follows this change improving the user experience.

About the Fibonacci, I decided to keep both numbers 1 to have an index match. With this, we match Fib(10) = space[10] = 55. What do you think about it?

**Comment by RayzaOliveira** [comment]
Cool, I really liked the proposal, because I believe that following this pattern would be simple to add a new font size in the future. Unlike the size array, where we would have to do a big refactoring if we needed to add a new font size.

About the spacings, I think we can follow this pattern without problems.

**Comment by arantespp** [comment]
I suggest we use the following array for breakpoints:

```ts
  /**
   * For a font size equals 16px, the breakpoints will be:
   * [640, 768, 1024, 1280, 1536, 1920] in pixels.
   */
  breakpoints: ['40rem', '48rem', '64rem', '80rem', '96rem', '120rem'],
```
